src/model/Stroke.cpp

```diff
@@ -137,16 +165,23 @@ void Stroke::setLastPoint(const Point& p) {
 
 void Stroke::addPoint(const Point& p) {
     this->points.emplace_back(p);
-    this->sizeCalculated = false;
+    updateBounds(Element::x, Element::y, Element::width, Element::height, Element::snappedBounds, p,
+                 hasPressure() ? p.z / 2.0 : this->width);
 }
 
 auto Stroke::getPointCount() const -> int { return this->points.size(); }
 
 auto Stroke::getPointVector() const -> std::vector<Point> const& { return points; }
 
-void Stroke::deletePointsFrom(int index) { points.resize(std::min(size_t(index), points.size())); }
+void Stroke::deletePointsFrom(int index) {
+    points.resize(std::min(size_t(index), points.size()));
```

Comment:
In principle in case `index > points.size()` it is not necessary to invalidate the cache. Not sure that is relevant though.

Reply:
Yes, that's right, but using an index, larger than `points.size() - 1` can be assumed to be a programming error.
So you are also right with point 2:
> Not sure that is relevant though.